styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java

```diff
@@ -547,19 +550,30 @@ private void setupMetrics(
                   .count()));
       stats.registerActiveStates(
           state,
-          "none",
-          () -> stateManager.activeStates().values().stream()
+          "none", () -> stateManager.activeStates().values().stream()
               .filter(runState -> runState.state().equals(state))
               .filter(runState -> !runState.data().trigger().isPresent())
               .count());
     });
 
-    stats.registerQueuedEvents(queuedEventsCount);
-    stats.registerWorkflowCount("all", allWorkflowsCount);
-    stats.registerWorkflowCount("configured", configuredWorkflowsCount);
-    stats.registerWorkflowCount("enabled", configuredEnabledWorkflowsCount);
-    stats.registerWorkflowCount("docker_termination_logging_enabled",
-                                dockerTerminationLoggingEnabledWorkflowsCount);
+    final Function<String, Gauge<Long>> createResourceCountGauge = (resource) -> {
+      final Supplier<Optional<Resource>> resourceSupplier =
+          new CachedSupplier<>(() -> storage.resource(resource), Instant::now);
+      return () -> resourceSupplier.get().map(Resource::concurrency).orElse(0L);
+    };
+
+    try {
+      storage.resources().forEach(resource -> stats
+          .registerResourceCount(resource.id(), createResourceCountGauge.apply(resource.id())));
+    } catch (IOException e) {
+      LOG.warn("Failed to get resources, skip registering resource count");
+    }
+
+    final Supplier<Optional<Long>> globalConcurrency =
+        new CachedSupplier<>(storage::globalConcurrency, Instant::now);
+    stats.registerResourceCount(Scheduler.GLOBAL_RESOURCE_ID, () ->
```

Comment:
Feels like a bigger change. I remember there was a concern having it the same as normal resource, could be overwritten or something by any user from API. We had two PRs to choose from and ended up choosing this approach.